### PR TITLE
[ETCM-943] partially split reading part of Blockchain

### PR DESCRIPTION
**Pull request by AurelienRichez**
# Description

The goal of this PR is to put the read only part of `Blockchain` in another class in order to reduce the dependency on this class. As a target I choose to make `BlockchainHostActor` independent of `Blockchain`. 

I moved a few function inside a new class named `BlockchainReader` for consistency, but we could come up with another name. 

It allowed to remove `Blockchain` at some other places but there are still read only methods that can be moved to go further. But as it is already a big change we should at least validate this before continuing.

# review

This is a big change, but mostly moving things around. Each commit can be viewed one by one and is mergeable, so we can split this PR if needed

One important thing to think about is : does the change help us for a future refactoring ? I don't necessarily see `BlockchainReader` as a final target for the structure of the code, but more as step to help us refactor step by step.

### CODE REVIEW

File: src/it/scala/io/iohk/ethereum/txExecTest/util/DumpChainApp.scala [hunk 1]
```diff
@@ -32,10 +32,17 @@ import io.iohk.ethereum.security.SecureRandomBuilder
 import io.iohk.ethereum.utils.{Config, NodeStatus, ServerStatus}
 import monix.reactive.Observable
 import org.bouncycastle.util.encoders.Hex
+import org.scalamock.clazz.Mock
```
Comment (leo-bogastry) [inline src/it/scala/io/iohk/ethereum/txExecTest/util/DumpChainApp.scala]: With `MockFactory` you shouldn't need this

File: src/main/scala/io/iohk/ethereum/domain/BlockchainReader.scala [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+import io.iohk.ethereum.db.storage.{
+  BlockBodiesStorage,
+  BlockHeadersStorage,
+  BlockNumberMappingStorage,
+  ReceiptStorage,
+  StateStorage
+}
+import io.iohk.ethereum.mpt.MptNode
+
+class BlockchainReader(
```
Comment (dzajkowski) [inline src/main/scala/io/iohk/ethereum/domain/BlockchainReader.scala:13]: would the next step be to create a trait `BlockchainReader` to allow for easier stubbing?

**Review by dzajkowski — APPROVED** [review]

File: src/main/scala/io/iohk/ethereum/domain/BlockchainReader.scala (hunk 2 above)
Comment (AurelienRichez) [reply src/main/scala/io/iohk/ethereum/domain/BlockchainReader.scala:13]: Yes I agree. In the end we might split it further and have several reader traits.

File: src/it/scala/io/iohk/ethereum/txExecTest/util/DumpChainApp.scala (hunk 1 above)
Comment (AurelienRichez) [reply src/it/scala/io/iohk/ethereum/txExecTest/util/DumpChainApp.scala]: indeed

**Review by leo-bogastry — APPROVED** [review]

**Review by AnastasiiaL — APPROVED** [review]

**Pull request merged**